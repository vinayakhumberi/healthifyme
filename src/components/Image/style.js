import styled from 'styled-components';
import { rem } from '../../utils/style.util';
export const Img = styled.img`
  width: ${props =>
    typeof props.width === 'string' ? props.width : rem(props.width)};
  height: ${props =>
    typeof props.height === 'string' ? props.height : rem(props.height)};
  padding-right: ${props =>
    typeof props.paddingRight === 'string'
      ? props.paddingRight
      : rem(props.paddingRight)};
  padding-left: ${props =>
    typeof props.paddingLeft === 'string'
      ? props.paddingLeft
      : rem(props.paddingLeft)};
  padding-bottom: ${props =>
    typeof props.paddingBottom === 'string'
      ? props.paddingBottom
      : rem(props.paddingBottom)};
  padding-top: ${props =>
    typeof props.paddingTop === 'string'
      ? props.paddingTop
      : rem(props.paddingTop)};
  object-fit: ${props => props.objectFit};
`;