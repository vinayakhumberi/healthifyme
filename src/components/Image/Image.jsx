import React from "react";
import PropTypes from "prop-types";
import { Img } from "./style";

const Image = function (props) {
  return (
    <Img
      data-test="image-container"
      {...props}
      src={props.src}
      alt={props.alt}
    />
  );
};
Image.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  paddingRight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  paddingLeft: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  paddingTop: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  paddingBottom: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  objectFit: PropTypes.string,
};

Image.defaultProps = {
  src:
    "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAAAyCAQAAAAAPLY1AAAAQ0lEQVR42u3PMREAAAgEID++qTWDqwcNSE+9EBERERERERERERERERERERERERERERERERERERERERERERERERGRiwXRD0zD8qRK9QAAAABJRU5ErkJggg==",
  alt: "zotribe",
  width: "auto",
  height: "auto",
  paddingRight: "inherit",
  paddingLeft: "inherit",
  paddingTop: "inherit",
  paddingBottom: "inherit",
  objectFit: "initial",
};

export default Image;
