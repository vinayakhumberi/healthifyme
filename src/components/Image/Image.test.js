import React from 'react';
import { shallow } from 'enzyme';
import Image from './Image';

describe('<Image />', () => {
  let wrapper;
  let imgUrl ="https://s3.ap-south-1.amazonaws.com/tpng-images/random/original/6be52f0b-56bf-4fdc-abd3-0bec9e58610c.png";
  let altText="tests";
  beforeEach(() => wrapper = shallow(<Image src={imgUrl} alt={altText} />));
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should Render a Image Container', () => {
    expect(wrapper.find('[data-test="image-container"]').length).toEqual(1);
  });
  it('Should pass src prop to Image', () => {
    expect(wrapper.find('[data-test="image-container"]').get(0).props.src).toEqual(imgUrl);
  });
  it('Should pass alt prop to Image', () => {
    expect(wrapper.find('[data-test="image-container"]').get(0).props.alt).toEqual(altText);
  });
});


describe('<Image />', () => {
  let wrapper;
  let imgUrl = "https://s3.ap-south-1.amazonaws.com/tpng-images/random/original/6be52f0b-56bf-4fdc-abd3-0bec9e58610c.png";
  let altText = "tests";
  const props = {
    width: "auto",
    height: "auto",
    paddingRight: "inherit",
    paddingLeft: "inherit",
    paddingTop: "inherit",
    paddingBottom: "inherit",
    objectFit: "initial",
  };
  beforeEach(() => wrapper = shallow(<Image src={imgUrl} alt={altText} {...props} />));
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should Render a Image Container', () => {
    expect(wrapper.find('[data-test="image-container"]').length).toEqual(1);
  });
  it('Should pass src prop to Image', () => {
    expect(wrapper.find('[data-test="image-container"]').get(0).props.src).toEqual(imgUrl);
  });
  it('Should pass alt prop to Image', () => {
    expect(wrapper.find('[data-test="image-container"]').get(0).props.alt).toEqual(altText);
  });
  it('Should pass width prop to Image', () => {
    expect(wrapper.find('[data-test="image-container"]').get(0).props.width).toEqual(props.width);
  });
  it('Should pass height prop to Image', () => {
    expect(wrapper.find('[data-test="image-container"]').get(0).props.height).toEqual(props.height);
  });
  it('Should pass paddingRight prop to Image', () => {
    expect(wrapper.find('[data-test="image-container"]').get(0).props.paddingRight).toEqual(props.paddingRight);
  });
  it('Should pass paddingLeft prop to Image', () => {
    expect(wrapper.find('[data-test="image-container"]').get(0).props.paddingLeft).toEqual(props.paddingLeft);
  });
  it('Should pass paddingTop prop to Image', () => {
    expect(wrapper.find('[data-test="image-container"]').get(0).props.paddingTop).toEqual(props.paddingTop);
  });
  it('Should pass paddingBottom prop to Image', () => {
    expect(wrapper.find('[data-test="image-container"]').get(0).props.paddingBottom).toEqual(props.paddingBottom);
  });
  it('Should pass objectFit prop to Image', () => {
    expect(wrapper.find('[data-test="image-container"]').get(0).props.objectFit).toEqual(props.objectFit);
  });
});
