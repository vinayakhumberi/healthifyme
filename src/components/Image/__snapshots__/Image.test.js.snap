// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders correctly 1`] = `ShallowWrapper {}`;

exports[`<Image /> renders correctly 2`] = `ShallowWrapper {}`;
